Refactor side nav and page layout

johnclary opened the pull request:
## Associated issues

* https://github.com/cityofaustin/atd-data-tech/issues/24098

This PR addresses a number of layout issues I encountered when working on the crashes map. Sorry if the commits look a bit odd—I extracted this work from https://github.com/cityofaustin/vision-zero/pull/1824. The need for some of these changes will be more obvious once I wrap up that PR.

## Summary of changes

### Top nav

* Now 100% width and fixed position
* Environment banner shows as a badge in side the navbar

### Side navigation

* Now has a light background when user is in "light" mode
* The active navigation item is highlighted in a lighter color
* There are text and background color effects when hovering over navigation items
* Uses fixed positioning

<img width="457" height="689" alt="sidenav-dark" src="https://github.com/user-attachments/assets/9a2777fb-e160-4b90-9419-e4008c34ee1d" />
<img width="617" height="803" alt="sidenav-light" src="https://github.com/user-attachments/assets/203e1ad4-064f-40af-980c-8876d562f275" />

### App container

* Now uses nested flexbox containers with `flex-grow-1` to ensure full-height
* The entire app now scrolls instead of the inner container scrolling only

### List pages

* Outer card has been removed

### List view table controls

* Changed the filter menu accordion to a dropdown
* Re-arranged the pagination controls
* Removed labels and added tooltips that appear on hover

<img width="389" height="71" alt="pagination-new" src="https://github.com/user-attachments/assets/71756e92-7763-4943-8620-44b7f1fb4407" />


<img width="1499" height="857" alt="crashes-light" src="https://github.com/user-attachments/assets/f98d054f-641f-44e0-9d6f-a18bb1290c5c" />


## Testing

**URL to test:** [Netlify](https://deploy-preview-1830--atd-vze-staging.netlify.app/)

1. In **light** and **dark modes**, visit all of the pages in the app and make sure they look ok ✨

---

#### Ship list

- [ ] Check migrations for any conflicts with latest migrations in `main` branch
- [ ] Confirm Hasura role permissions for necessary access
- [ ] Code reviewed
- [ ] Product manager approved

johnclary commented on editor/components/TableWrapper.tsx:
```diff
@@ -84,9 +82,30 @@ export default function TableWrapper<T extends Record<string, unknown>>({
+  /**
+   * Array of columns which should be fetched - this is
+   * is the combination of visibile columns + fetchAlways columns
+   */
```
this new `fetchAlways` setting came out of my map work, specifically to handle when we want to include `latitude` and `longitude` columns in our query but we don't need to show them in the table or expose them in the col picker.

johnclary commented on editor/components/TableResetFiltersToggle.tsx:
```diff
@@ -5,7 +5,8 @@ import AlignedLabel from "./AlignedLabel";
 
 interface Props {
-  queryConfig: QueryConfig;
+  isMapActive: boolean;
```
this prop will be implemented on the map view PR that bases against this

chiaberry commented on editor/components/EnvironmentBadge.tsx:
I like this switch to a badge instead of a banner!

mddilley commented on editor/components/SideBarListItem.tsx:
```diff
@@ -22,19 +22,19 @@ export default function SideBarListItem({
     <ListGroup.Item
-      className={`mx-0 bg-dark fs-5 my-1 ${
-        isCurrentPage ? "text-white" : "text-secondary"
+      className={`sidebar-list-item fs-5 py-2 ps-4 ${
```
these style updates for the current page are great! ✨

mddilley commented on editor/components/TableAdvancedSearchFilterToggle.tsx:
I think interacting with the filters takes a bit more effort, but I also think that the filter count being prominent in the list pages helps with quickly seeing that filters are applied. I definitely like these tucked-in toggles more than taking over a large portion of the screen to show the filters - especially like in the EMS incidents table where there are many way to filter now!

It is a tough balance! I think that user feedback will come in handy here if any update is needed.

mddilley reviewed (approved):
I tested locally with multiple roles, compared to staging, and it all works so well. So many nice touches in here. 🚢 🚀 ✨

johnclary commented on editor/components/SidebarLayout.tsx:
```diff
@@ -13,36 +11,41 @@ import PermissionsRequired from "@/components/PermissionsRequired";
+    if (isCollapsed) {
+      const timeout = setTimeout(() => {
+        setIsCollapsed(false);
+      }, 250);
```
yeah that's what i was thinking! although maybe we could increase it a bit. a quarter of a second is pretty short!

johnclary commented on editor/components/SidebarLayout.tsx:
```diff
@@ -13,36 +11,41 @@ import PermissionsRequired from "@/components/PermissionsRequired";
 import AppBreadCrumb from "@/components/AppBreadCrumb";
 import AppFooter from "@/components/AppFooter";
 import { darkModelocalStorageKey } from "@/components/DarkModeToggle";
```
Fixed—thank you!

johnclary commented on editor/styles/global.scss:
Good to hear this—I will make this a top priority. Issue coming momentarily. See also: https://github.com/cityofaustin/vision-zero/pull/1830/files#r2288821226

johnclary commented on editor/styles/global.scss:
```diff
@@ -83,17 +100,84 @@ $dark: $night;
+  background-color: #495057 !important;
+}
+
+[data-bs-theme="light"] .sidebar-list-item:hover {
+  color: #000 !important;
+}
+
+[data-bs-theme="dark"] .sidebar-list-item:hover {
+  color: #fff !important;
+}
+
```
* https://github.com/cityofaustin/atd-data-tech/issues/24266

roseeichelmann reviewed (approved):
i lovee these changes! the feedback for hovering on the side nav options is chefs kiss, also great that you added labels to them on hover bc sometimes it was guesswork trying to remember which symbol went to which page.  i also love all the tool tips and simplifying the labels on the table wrapper component.

my only feedback that is maybe in scope for this PR is how squished the top nav gets on small screen sizes:

This PR:
<img width="886" height="1279" alt="Screenshot from 2025-08-20 15-44-15" src="https://github.com/user-attachments/assets/a0d6a54e-5465-4245-bb86-6bee8ebdb8a3" />


Production:
<img width="886" height="1279" alt="Screenshot from 2025-08-20 15-44-22" src="https://github.com/user-attachments/assets/55d61f55-fba1-47e3-815c-12c0fcb4048f" />

johnclary commented:
@roseeichelmann i added a display class to the navbar search to hide it at `xs` and `sm`. i also added some `flex-nowrap` classes to keep a few bits of the table search from awkwardly wrapping.

there's more work to do on this front. I opened #24283.

<img width="552" height="343" alt="Screenshot 2025-08-21 at 10 30 03 AM" src="https://github.com/user-attachments/assets/fc461505-f1d1-402e-ad05-8b8fab8fb402" />
